src/proxy.rs

```diff
@@ -107,17 +109,23 @@ pub fn filter_frontend_request_headers(headers: &Headers) -> Headers {
 ///
 /// The primary purpose of this function is to add and remove headers as required by an
 /// intermediary conforming to the HTTP spec.
-fn map_request(req: server::Request, url: &str) -> client::Request {
-    let mut r = client::Request::new(req.method().clone(), url.parse().unwrap());
-
+fn map_request(req: server::Request, url: Url) -> client::Request {
     let via = create_via_header(
         req.headers().get::<Via>(),
         req.version());
 
     let mut headers = filter_frontend_request_headers(req.headers());
     headers.set(via);
-    r.headers_mut().extend(headers.iter());
 
+    // add host header related to backend
```

Comment:
This was necessary to support heroku apps:
```
cargo run --bin weldr 127.0.0.1:9001 https://wiki-growth.herokuapp.com/ 127.0.0.1:9002
```
The browser is sending requests to http://127.0.0.1:9001 (`Host = 127.0.0.1` )
When heroku receives those requests, it does not know on which applications the request should be send to.

Thinking twice about it, translating the `Host` header is maybe not the main use-case.
We should make this configurable.

For example, nginx makes this configurable:
```
server {
  listen 80;
  server_name example.com;
  location / {
    proxy_pass       http://main;
    proxy_set_header Host            $host;
    proxy_set_header X-Forwarded-For $remote_addr;
  }
}
```